analyses/molecular-subtyping-EWS/01-run-subtyping-ewings.Rmd

```diff
@@ -0,0 +1,120 @@
+---
+title: "Add molecular subtype for samples that have hallmark Ewings Sarcoma fusions"
+output: html_notebook
+author: K S Gaonkar for D3b
+date: January 2020
+---
+
+Identify sample IDs with hallmark _EWSR1_ fusions and subtype as `EWS` 
+
+```{r}
+library("tidyverse")
+```
+
+### Read in data
+
+```{r}
+# to get root directory
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# data directory
+dataDir <- file.path(root_dir, "data")
+
+# histology file
+clinical <- readr::read_tsv(file.path(dataDir, "pbta-histologies.tsv"),
+                            guess_max = 10000) %>%
+  select(Kids_First_Biospecimen_ID,
+         Kids_First_Participant_ID, 
+         experimental_strategy, 
+         sample_id,
+         sample_type)
+
+# get ewings fusion calls from fusion-summary results file
+ewings_foi_bsids <- readr::read_tsv(
+  file.path(root_dir, 
+            "analyses",
+            "fusion-summary",
+            "results",
+            "fusion_summary_ewings_foi.tsv")
+  ) 
+
+# get subset folder
+results_dir <- "results"
+
+# create if doesn't exist
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+```
+
+### Add "EWS" to molecular_subtype for samples that have hallmark ewings sarcoma EWSR1 fusion in RNA and match to DNA per sample_id
+
+```{r}
+# get a vector of biospecimen IDs for those biospecimens that should be subtyped
+have_ews_fusion <- rowSums(ewings_foi_bsids[, -1]) > 0
+
+# get rna BS_IDs which have hallmark fusion 
+ews_rnaseq_bsids <- ewings_foi_bsids %>%
+  filter(have_ews_fusion) %>%
+  pull(Kids_First_Biospecimen_ID)
+
+# Get RNAseq ids because we have fusion information for RNA-seq samples
+ews_rnaseq <- clinical %>%
+  filter(Kids_First_Biospecimen_ID %in% ews_rnaseq_bsids,
+         sample_type == "Tumor") %>%
+  select(Kids_First_Biospecimen_ID, 
+         Kids_First_Participant_ID, 
+         sample_id) 
+
+# we're going to join the WGS samples to RNA-seq with sample_id
+# so that the WGS biospecimens that correspond to RNA-seq specimens 
+# can be merged in the next step
+ews_wgs <- clinical %>%
+  filter(sample_id %in% ews_rnaseq$sample_id,
+         sample_type == "Tumor",
+         experimental_strategy == "WGS") %>%
+  select(Kids_First_Biospecimen_ID, 
+         Kids_First_Participant_ID, 
+         sample_id)
+
+# Join together
+ews_subtype <- ews_rnaseq %>% 
+  full_join(ews_wgs,
+             by = c("Kids_First_Participant_ID",
+                    "sample_id"),
+             suffix = c("_RNA", "_DNA")) %>%
+  select(Kids_First_Participant_ID,
+         Kids_First_Biospecimen_ID_RNA,
+         Kids_First_Biospecimen_ID_DNA,
+         sample_id,
+         everything()) %>%
+  dplyr::mutate(molecular_subtype="EWS")
+```
+
+
+#### Save to file
+
+```{r}
+ews_subtype %>%
+  readr::write_tsv(file.path(results_dir, "EWS_results.tsv"))
+```
+
+### List which samples are molecular_subtype == "EWS"
```

Comment:
You may want to do these previews before you write to file. In other words, if something in these previews informs us about our resulting output we might want to know that before we write to file.